### ISSUE TITLE
provenance: know what a container holds, not just what an expression evaluates to

### ISSUE THREAD

**Issue by HimanshuJanbandhu** [highwaterlabs/torch-preflight#55]
The one `pytorch/tutorials` finding that within-file return resolution (#49) does not clear.

`chatbot_tutorial.py`:

```python
print_losses = []
...
print_losses.append(mask_loss.item() * nTotal)   # floats go in
...
return sum(print_losses) / n_totals              # a float comes out
```

The caller then does `print_loss += loss`, and TG001 reports a retained graph.

#49 taught the analysis to read a local helper's `return`, but it only accepts a **visible** detaching call in the returned expression — deliberately, since #46 established that "we could not prove it carries a graph" must never be read as "it does not". Here the `.item()` is one hop deeper, inside the list *elements*, so there is nothing visible in `sum(print_losses) / n_totals`.

### What it needs

Element types for containers: knowing that everything appended to `print_losses` was scalarised, so a reduction over it is too. That is type inference over a container rather than expression evaluation, which is why it was not bundled into #49.

### Why it is filed rather than done

**One finding is not yet a pattern.** It is worth building when the shape shows up in more than one place — the corpus scan in #56 is the thing that would tell us. Filed now so the evidence is not lost.

Test fixture to reduce from: `tutorials/beginner_source/chatbot_tutorial.py:1061`.

Whatever is built here must keep the guard from #46 — see `test_tg001_absence_of_proof_is_not_evidence_of_detachment`.
